server/datastore/mysql/microsoft_mdm.go

```diff
@@ -666,6 +666,7 @@ SELECT
 		WHEN (%s) THEN '%s'
 		WHEN (%s) THEN '%s'
 		WHEN (%s) THEN '%s'
+		ELSE ''
```

Comment:
do we know when/why this happens? 🤔

Reply:
I found it can happen when this method is called for Windows server (same as the case that is mocked in the test).

Reply:
thanks, this strikes me as something we should let fail in that case? and fix the root cause (not send commands to encrypt servers)? otherwise the problem will be hidden, right?

Reply:
It isn't limited to fleetd operations though. This method is getting called when the UI requests the host details page, leading to a 500 that renders the host details page inaccessible. We have a follow up [ticket](https://github.com/fleetdm/fleet/issues/17481) open to address the root cause.

Reply:
it's that something doesn't add up... aren't we checking if the host is a server right above? shouldn't fixing the root cause fix this as well?

Reply:
You're right, I bet it is another case where those `MDMInfo` methods are tripping us up!

Reply:
Yep, that's the issue alright. I'll open a new PR.